pandas/core/arrays/arrow/array.py

```diff
@@ -890,25 +890,54 @@ def _op_method_error_message(self, other, op) -> str:
     def _evaluate_op_method(self, other, op, arrow_funcs) -> Self:
         pa_type = self._pa_array.type
         other_original = other
-        other = self._box_pa(other)
+        try:
+            other = self._box_pa(other)
+        except pa.lib.ArrowTypeError:
+            # was expecting time dtype but received non-temporal dtype (time offset)
+            from pandas.core.tools.timedeltas import to_timedelta
+
+            other = self._box_pa(to_timedelta(other))
+        except ValueError as err:
+            raise TypeError(
+                "Incompatible type when converting to PyArrow dtype for operation."
+            ) from err
 
         if (
             pa.types.is_string(pa_type)
             or pa.types.is_large_string(pa_type)
             or pa.types.is_binary(pa_type)
         ):
             if op in [operator.add, roperator.radd]:
-                sep = pa.scalar("", type=pa_type)
-                try:
-                    if op is operator.add:
-                        result = pc.binary_join_element_wise(self._pa_array, other, sep)
-                    elif op is roperator.radd:
-                        result = pc.binary_join_element_wise(other, self._pa_array, sep)
-                except pa.ArrowNotImplementedError as err:
+                # pyarrow gets upset if you try to join a NullArray
```

Comment:
none of this.  like i said before, just take the diff from my PR and update the one affected test.